### PR TITLE
Add `shard.yml`

### LINKED ISSUE #14363: Manage compiler dependencies with `shards`
The compiler has two external dependencies vendored in at `lib/`: `reply` and `markd`.
To keep these synced with upstream, we use `git subtree`. Or at least that was the plan. It was [never setup for `lib/reply`](https://github.com/crystal-lang/crystal/pull/13884#issuecomment-1757077159).
And while it's not terribly complex to setup and maintain, `git subtree` adds some overhead and it's not very straightforward to understand what's going on.

In the Crystal ecosystem we actually have a great tool for managing dependencies: `shards`!
Now of course the compiler should not *require* `shards` for building to avoid circular dependencies. But there's no need to. If we continue to keep the lib sources vendored in the repository, `shards` is only used for *updating* the sources. That would be a simple `shards update` then. Compare that with the commands for `git subtree` which I'm far from being able to recall from the top of my head. Of course, that could be automated. But the point stands that `shards` is just so much easier to use. And you have the full features available to check which versions are installed, check if they're outdated etc.
We should expect compiler dependencies to be published as Crystal shards, so this should work with any future additions.

So here's what I'm proposing:
* Put a `shard.yml` into this repository with declared dependencies of `reply` and `markd` at the currently installed versions.
* Run `shards install` and there should be no difference in the `lib/` folder except a new `lib/.shards.info` file (which will also be checked in like the rest of `lib/`, and `shard.lock` as well)

### PR DESCRIPTION
**Pull request by straight-shoota**
Resolves #14363

### CODE REVIEW

**Comment by HertzDevil** [comment]
When I tried this, the symlink directories at `lib/markd/lib` and `lib/reply/lib` were included in the commit, so I wonder if something went wrong in my setup

**Comment by straight-shoota** [comment]
Ah no, I messed up with my git diff. It looked clean when it wasn't. 🤦 The lib symlinks are to be included, of course.

File: shard.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: crystal
+version: 1.12.0-dev
+
+authors:
+- Crystal Core Team <crystal@manas.tech>
+
+descriptin: |
```
Comment (Sija) [inline shard.yml:7]:
```suggestion
description: |
```

File: shard.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  markd:
+    github: icyleaf/markd
+    commit: 5e5a75d13bfdc615f04cc7ab166ee279b3b996d3
+  reply:
+    github: I3oris/reply
+    commit: 90a7eb5a76048884d5d56bf6b9369f1e67fdbcd7
+
+license: Apache-2.0
+
+repository: "https://github.com/crystal-lang/crystal"
```
Comment (Sija) [inline shard.yml:22]:
```suggestion
repository: https://github.com/crystal-lang/crystal
```

**Review by ysbaddaden — APPROVED** [review]

**Pull request merged**